apps/viewer/src/app.tsx

```diff
@@ -66,14 +71,24 @@ function App() {
 
   useEffect(() => {
     if (mainSourceId) {
-      projectToMainStream(mainSourceId);
+      // Change main stream
+      projectToMainStream(mainSourceId).then((source) => {
+        if (!source) {
+          return;
+        }
+
+        // Reapply previous quality setting
```

Comment:
Sometimes re-apply the previous quality setting could be failed because of the stream projection delay, I suggest resetting to `auto` when user switches stream.